intake_aodn/drivers.py

```diff
@@ -72,18 +72,25 @@ def __init__(self,
         self._ds = None
         self._load_data = False
         super(RefZarrStackSource, self).__init__(metadata=metadata, **kwargs)
-
+        
+    def clean_attrs(self,ds):
+        """ remove some attrs that prevent simple save to netcdf """
+        for v in ds.variables:
+            for bad_attr in ['_Netcdf4Dimid','NAME']:
+                if bad_attr in ds[v].attrs.keys():
+                    del ds[v].attrs[bad_attr]
+        return ds
     def to_dask(self):
         """Return xarray object where variables are dask arrays"""
         self._load_data = False
         self._load_metadata()
-        return self._ds
+        return self.clean_attrs(self._ds)
 
     def read(self):
         """Return xarray object where variables are dask arrays"""
         self._load_data = True
         self._load_metadata()
-        return self._ds
+        return self.clean_attrs(self._ds)
```

Comment:
Rather than call clean_attrs twice, I would call it just before assignment to self._ds during the open dataset. Also doesn't really make sense to pass self.ds to a method which operates on an instance of an object as you could just reference the 'private' self._ds.